### ISSUE TITLE
Render BlockExpression differently in single-statement context (e.g. test part for IfThen / IfThenElse)

### ISSUE THREAD

**Issue by zspitz** [zspitz/ExpressionToString#58]
For C#:
```csharp
if (testStatement1(), testStatement2()) {
}
```
instead of:
```csharp
if ({
    testStatement1();
    testStatement2();
}) {
}
```
like every other rendering of `BlockExpression`.

For VB.NET:
```vb
If testStatement1 : testStatement2 Then
End If
```
```vb
If
    testStatement1
    testStatement2
Then
End If
````
The use case is slightly more compelling in C#, where the `,` operator is a valid means of combining statements. VB.NET has no such construct.

**Comment by zspitz** [comment]
Final behavior as of e5c4217:

For C#:
```csharp
// blocks within inline expressions
() => (
    true,
    true
) ? true : false

// blocks within test clauses
if (
    true,
    true
) {
    Console.WriteLine(true);
}

// blocks directly within syntactic blocks
if (true) {
    Console.WriteLine(true);
    Console.WriteLine(true);
}

// nested blocks are only rendered if the inner block introduces new variables 
```
---
For VB:

```vb
' inside of an expression -- explicit Block construct
If Block
    True
    True
End Block Then Console.WriteLine

If True Then
    Console.WriteLine
    Console.WriteLine
End If

' for nested blocks the inner one is only rendered using Block...End Block if it introduces new variables
```

**Comment by zspitz** [comment]
Fixed by e5c4217

**Comment by zspitz** [comment]
Nested block behavior is still broken. The following should be happening:
```csharp
// blocks within inline expressions -- no variable
() => (
    true,
    true,
    true,
    true
) ? true : false

// blocks within inline expressions -- with variable
() => (
    true,
    (
        string s,
        true,
        true
    ),
    true
) ? true : false

// blocks within test clauses -- no variable
if (
    true,
    true,
    true,
    true
) {
    Console.WriteLine(true);
}

// blocks within test clauses -- with variable
if (
    true,
    (
        string s,
        true,
        true
    ),
    true
) {
    Console.WriteLine(true);
}

// blocks directly within syntactic blocks -- no variable
if (true) {
    Console.WriteLine(true);
    Console.WriteLine(true);
    Console.WriteLine(true);
    Console.WriteLine(true);
}

// blocks directly within syntactic blocks -- with variable
if (true) {
    Console.WriteLine(true);
    {
        string s;
        Console.WriteLine(true);
        Console.WriteLine(true);
    }
    Console.WriteLine(true);
}

```
